tests/security/pam/pam_mount.pm

```diff
@@ -0,0 +1,127 @@
+    my $loop_vol = 'enc_loop';
+
+    # Setup a loop device to be used as encrypted volume
+    assert_script_run "dd if=/dev/zero of=$testfile bs=1024k count=500";
+    assert_script_run "losetup $loopdev $testfile";
+    assert_script_run "losetup -a | grep $testfile";
+    script_run_interactive(
+        "cryptsetup luksFormat --type luks2 $loopdev",
+        [
+            {
+                prompt => qr/Are you sure.*/m,
+                string => "YES\n",
+            },
+            {
+                prompt => qr/Enter passphrase.*/m,
+                string => "$key\n",
+            },
+            {
+                prompt => qr/Verify passphrase.*/m,
+                string => "$key\n",
+            },
+        ],
+        400
+    );
+
+    # Create file "/etc/pam_mount_keys/enc_key" with a password contained
+    my $key_dir  = '/etc/pam_mount_keys';
+    my $key_file = 'enc_key';
+    assert_script_run "mkdir -p $key_dir";
+    assert_script_run "echo '$key' > $key_dir/$key_file";
+    assert_script_run(
+        "expect -c 'spawn cryptsetup luksAddKey $loopdev $key_dir/$key_file; \\
+expect \"Enter any existing passphrase: \"; send \"$key\\n\"; interact'"
+    );
+    assert_script_run(
+        "expect -c 'spawn cryptsetup luksOpen $loopdev $loop_vol; \\
+expect \"Enter passphrase for \/dev\/$loopdev: \"; send \"$key\\n\"; interact'"
+    );
+    assert_script_run "mkfs.ext4 -L $user /dev/mapper/$loop_vol";
+    assert_script_run "cryptsetup luksClose $loop_vol";
+
+    # Configure the "/etc/security/pam_mount.conf.xml" file
+    my $pam_mount_cfg     = '/etc/security/pam_mount.conf.xml';
+    my $pam_mount_cfg_bak = '/etc/security/pam_mount.conf.xml.bak';
+    assert_script_run "cp $pam_mount_cfg $pam_mount_cfg_bak";
+    assert_script_run "sed -i '/<pam_mount>/,/<\\/pam_mount>/d' $pam_mount_cfg";
+    assert_script_run(
+        "echo \"\$(cat <<EOF
+<pam_mount>
+  <volume user=\"$user\" path=\"$loopdev\" mountpoint=\"~\" fstype=\"crypt\" fskeycipher=\"none\" fskeyhash=\"md5\" fskeypath=\"$key_dir/$key_file\" />
+</pam_mount>
+EOF
+        )\" >> $pam_mount_cfg"
+    );
+
+    # Modify the pam common-session and common-auth files
+    my $pam_session     = '/etc/pam.d/common-session';
+    my $pam_session_bak = '/etc/pam.d/common-session.bak';
+    my $pam_auth        = '/etc/pam.d/common-auth';
+    my $pam_auth_bak    = '/etc/pam.d/common-auth.bak';
+    assert_script_run "cp $pam_session $pam_session_bak";
+    assert_script_run "cp $pam_auth $pam_auth_bak";
+    assert_script_run "sed -i '\$a session \[success=1 default=ignore\] pam_succeed_if.so service = systemd-user' $pam_session";
+    assert_script_run "sed -i '\$a session optional        pam_mount.so   disable_interactive' $pam_session";
+    assert_script_run "sed -i '\$a auth \[success=1 default=ignore\] pam_succeed_if.so service = systemd-user' $pam_auth";
+    assert_script_run "sed -i '\$a auth    optional        pam_mount.so   disable_interactive' $pam_auth";
```

Comment:
There are lots of `sed -i '\$a ...` what is the meaning of `$a` ?

Reply:
The last line, any better code style here? pleas suggest

Reply:
No suggestion I just question.

Reply:
Btw, I mean upload "$pam_session/$pam_auth/..." for reference but it is optional and depends on you.

Reply:
done